### ISSUE TITLE
AzureMonitor: Scope error in configuration UI when legacy "same as" is false

### ISSUE THREAD

**Issue by kostrse** [grafana/grafana#37134]
**What happened**:

Error message about invalid scope when Azure Monitor datasource has different credentials for Metrics and Log Analytics.
Didn't see this error for datasources with only Metrics credentials.

![image](https://user-images.githubusercontent.com/1131253/126729090-0ef32218-2a5a-4f47-84d8-5d9c806b85a2.png)

Error message:

> unexpected error Get "https://management.azure.com/subscriptions?api-version=2019-03-01": failed to retrieve Azure access token: invalid_scope AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope https://management.azure.com is not valid. Trace ID: d69cadf9-38a4-4bd5-bdd2-60a0b9612c00 Correlation ID: 1aedb56a-0312-4598-9def-cf85702bbb52 Timestamp: 2021-07-23 01:57:07Z

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

To existing working Azure Monitor datasource, add the following additional fields into JsonData:
```json
{
    "azureLogAnalyticsSameAs": false,
    "logAnalyticsTenantId": "<same as tenantId>",
    "logAnalyticsClientId": "<same as clientId>"
}
```

Configuration should be similar to this:

**JsonData:**
```json
{
    "azureAuthType": "clientsecret",
    "cloudName": "azuremonitor",
    "tenantId": "aaa",
    "clientId": "bbb",

    "azureLogAnalyticsSameAs": false,

    "logAnalyticsTenantId": "aaa",
    "logAnalyticsClientId": "bbb"
}
```

**SecureJsonData:**
```json
{
    "clientSecret": "..."
}
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: main
- Grafana plugins: Azure Monitor

**Comment by kostrse** [comment]
I was able to consistently reproduce it today in earlier version of `grafana/grafana:main` Docker container but it seems gone on the most latest build.

I will retest it on `8.1-beta2` and close if not reproducible..

**Comment by andresmgot** [comment]
can it be that the "Default Subscription" field is empty on your first test?

**Comment by kostrse** [comment]
Fixed #37981
